modules/brushingandlinking/src/brushingandlinkingmanager.cpp

```diff
@@ -394,12 +402,18 @@ int BrushingAndLinkingManager::getActionIndex(BrushingAction action) {
 void BrushingAndLinkingManager::propagate(BrushingAction action, BrushingTarget target) {
     modifications_[target] |= fromAction(action);
 
-    if (std::holds_alternative<BrushingAndLinkingOutport*>(owner_)) {
-        auto outport = std::get<BrushingAndLinkingOutport*>(owner_);
-        outport->getProcessor()->invalidate(invalidationLevel_);
-    } else if (!parent_) {
-        auto inport = std::get<BrushingAndLinkingInport*>(owner_);
-        inport->getProcessor()->invalidate(invalidationLevel_);
+    // Only invalidate the top level in the connected brushing manager network
+    if (!parent_) {
+        if (std::holds_alternative<BrushingAndLinkingOutport*>(owner_)) {
+            auto outport = std::get<BrushingAndLinkingOutport*>(owner_);
+            // Processor need to be invalidated for the network evaluation to be notified.
+            outport->getProcessor()->invalidate(invalidationLevel_);
+        } else if (std::holds_alternative<BrushingAndLinkingInport*>(owner_)) {
+            // Nothing is connected, invalidate the processor itself
+            auto inport = std::get<BrushingAndLinkingInport*>(owner_);
+
+            inport->getProcessor()->invalidate(invalidationLevel_);
```

Comment:
```suggestion
            inport->invalidate(invalidationLevel_);
```
Should be the same, and is cleaner I think. I not the same, explain why.

Reply:
Yes, here I think it should be the same